tests/lib/rule-tester/flat-rule-tester.js

```diff
@@ -0,0 +1,2558 @@
+        ruleTester.run("no-test-settings", require("../../fixtures/testers/rule-tester/no-test-settings"), {
+            valid: [
+                {
+                    code: "var test = 'bar'", settings: { test: 1 }
+                }
+            ],
+            invalid: [
+                {
+                    code: "var test = 'bar'", settings: { "no-test": 22 }, errors: 1
+                }
+            ]
+        });
+    });
+
+    it("should pass-through the filename to the rule", () => {
+        (function() {
+            ruleTester.run("", require("../../fixtures/testers/rule-tester/no-test-filename"), {
+                valid: [
+                    {
+                        code: "var foo = 'bar'",
+                        filename: "somefile.js"
+                    }
+                ],
+                invalid: [
+                    {
+                        code: "var foo = 'bar'",
+                        errors: [
+                            { message: "Filename test was not defined." }
+                        ]
+                    }
+                ]
+            });
+        }());
+    });
+
+    it("should pass-through the options to the rule", () => {
+        ruleTester.run("no-invalid-args", require("../../fixtures/testers/rule-tester/no-invalid-args"), {
+            valid: [
+                {
+                    code: "var foo = 'bar'",
+                    options: [false]
+                }
+            ],
+            invalid: [
+                {
+                    code: "var foo = 'bar'",
+                    options: [true],
+                    errors: [{ message: "Invalid args" }]
+                }
+            ]
+        });
+    });
+
+    it("should throw an error if the options are an object", () => {
+        assert.throws(() => {
+            ruleTester.run("no-invalid-args", require("../../fixtures/testers/rule-tester/no-invalid-args"), {
+                valid: [
+                    {
+                        code: "foo",
+                        options: { ok: true }
+                    }
+                ],
+                invalid: []
+            });
+        }, /options must be an array/u);
+    });
+
+    it("should throw an error if the options are a number", () => {
+        assert.throws(() => {
+            ruleTester.run("no-invalid-args", require("../../fixtures/testers/rule-tester/no-invalid-args"), {
+                valid: [
+                    {
+                        code: "foo",
+                        options: 0
+                    }
+                ],
+                invalid: []
+            });
+        }, /options must be an array/u);
+    });
+
+    describe("Parsers", () => {
+
+        it("should pass-through the parser to the rule", () => {
+            const spy = sinon.spy(ruleTester.linter, "verify");
+            const esprima = require("esprima");
+
+            esprima.name = "esprima";
```

Comment:
```suggestion
```

Do we need to set `esprima.name`, since the parsers are compared by reference below?